.github/workflows/release.yml

```diff
@@ -1,49 +1,110 @@
 name: release
 
+# Automate deployment to PyPi when creating a release tag vX.Y.Z
+# will only be published to PyPi if the git tag matches the release version
+# and the pre-commit and tests pass
+
 on:
-  release:
-    types: [published, edited]
+  push:
+    tags:
+      - "v*"
```

Comment:
Should we make this slightly more specific? what if we create a tag `virtual-machine-tutorial-2024` at some point? Maybe `v[0-9]+.[0-9]+.[0-9]+*` would work?

Reply:
sounds reasonable, I will just give it a try in my test repo...

Reply:
yep works fine so changed